[3.3.5-proc] Unit::HandleAuraProc

ccrs opened the pull request:
@ariel- 

you are not alone bby

crsdev committed:
Gaseous Bloat

crsdev committed:
Deadly Precision

crsdev committed:
some corrections

Shauren commented on sql/updates/world/3.3.5/2016_99_99_99_world .sql:
```diff
@@ -0,0 +1,12 @@
+DELETE FROM `spell_proc` WHERE `SpellId` IN (70672, 72455, 72832, 72833, 71564);
```
dat space in filename

crsdev committed:
some corrections

crsdev committed:
Ball of Flames Proc

crsdev committed:
Discerning Eye of the Beast

ariel- commented on src/server/scripts/Northrend/IcecrownCitadel/boss_blood_prince_council.cpp:
```diff
@@ -1498,6 +1498,27 @@ class spell_taldaram_ball_of_inferno_flame : public SpellScriptLoader
+            void HandleStackDrop(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+            {
+                PreventDefaultAction();
+                GetTarget()->RemoveAuraFromStack(GetId());
```
ModStackAmount(-1);

ccrs replied:
no need to, that function already does that and in case there is no stack left it removes the aura... so yeah no need to change it

crsdev committed:
Swift Hand of Justice

ariel- commented on src/server/scripts/Spells/spell_item.cpp:
```diff
@@ -1757,6 +1757,49 @@ class spell_item_six_demon_bag : public SpellScriptLoader
+                PreventDefaultAction();
+
+                Unit* target = GetTarget();
+                int32 bp0 = CalculatePct(target->GetMaxHealth(), GetSpellInfo()->Effects[EFFECT_0].CalcValue());
```
Just use aurEff->GetAmount()

ariel- commented on src/server/scripts/Spells/spell_item.cpp:
```diff
@@ -333,6 +333,74 @@ class spell_item_blessing_of_ancient_kings : public SpellScriptLoader
+
+            void HandleDummy(SpellEffIndex /*effIndex*/)
+            {
+                SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(SPELL_DEADLY_PRECISION);
```
As the spellInfo exists, because of Validate hook, use AssertSpellInfo so Coverity doesn't complain

crsdev committed:
some corrections

crsdev committed:
some corrections

crsdev committed:
Infusion of Light

crsdev committed:
Judgements of the Just

crsdev committed:
some corrections

crsdev committed:
Glyph of Divinity

crsdev committed:
some corrections

crsdev committed:
Combustion

crsdev committed:
Empowered Fire

crsdev committed:
Ignite & some corrections

crsdev committed:
travis

crsdev committed:
Missile Barrage, Hot Streak & Fireball! extra effects

crsdev committed:
some corrections

crsdev committed:
Brain Freeze & corrections

crsdev committed:
Blood of the North & Reaping & Death Rune Mastery

crsdev committed:
some corrections

ariel- committed:
Fix stuff™:

- Add spell family flags for procs, prevents Blood of the North and friends from proccing off the wrong spells
- Combustion, same story. Apply buff in checkproc hook to prevent dropping charges
- Swift hand of Justice: must use triggered spell's base points instead of aura's bp
- Cleaned up some logic in death rune effects
- Removed Missile Barrage and friends from sql, this was already taken care of in 'base' sql
- Moved Mage T8 4P bonus handling to CheckProc hook, to prevent dropping charges

ariel- commented:
So, turns out the combustion issues were caused by the check you mentioned.

Mage casting times were another bug, and you included them in the sql even when I had event entries already :P

You were missing SpellFamilyFlags on several entries, which caused auras to incorrectly proc off other spells.

That being said, thanks for your help!

ccrs commented:
no problem at all, my pleasure

but next time dont delete my branch XD
